### PR TITLE
Usability fixes for examples

### PR DESCRIPTION
**Pull request by Roshrini**
## Description ##
Fixing examples- lstm, matrix_factorization, vae and adding readme file.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by Roshrini** [comment]
@aaronmarkham @sandeep-krishnamurthy @nswamy

File: example/model-parallel/lstm/README.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
 Model Parallel LSTM
 ===================
+
 This is an example showing how to do model parallel LSTM in MXNet.
-Most of the code is duplicated with the rnn example, and should be eventually merged.
+
+To run this example, first make sure you download a dataset of PenTreeBank available
```
Comment (sandeep-krishnamurthy) [inline example/model-parallel/lstm/README.md]: nitpik: We use the [PenTreeBank dataset](link)  in this example. Download the dataset with below command.

File: example/model-parallel/lstm/README.md [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+[here](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/ptb/) by running following command:
+
+`bash get_ptb_data.sh`
+
+This will download PenTreeBank dataset under `data` folder. Now, you can run the training as follows:
```
Comment (sandeep-krishnamurthy) [inline example/model-parallel/lstm/README.md]: the PenTreeBank

File: example/model-parallel/matrix_factorization/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Model Parallel Matrix Factorization
+===================================
+
+The example walks you through a matrix factorization algorithm for recommendations and also
```
Comment (sandeep-krishnamurthy) [inline example/model-parallel/matrix_factorization/README.md]: The example => This example.
Will be helpful to link the paper for matrix factorization algorithm

File: example/model-parallel/matrix_factorization/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+trained on cpu and the other on gpu.
+
+To run this example, first make sure you download a dataset of 10 million movie ratings available
+from [the MovieLens project](http://files.grouplens.org/datasets/movielens/) by running following command:
+
+`python get_data.py`
+
+This will download MovieLens 10M dataset under ml-10M100K folder. Now, you can run the training as follows:
+
+`python train.py --num-gpus 2`
```
Comment (sandeep-krishnamurthy) [inline example/model-parallel/matrix_factorization/README.md]: Also please add if only CPUs are available.
And, majority of users will have only 1 GPU, may be good idea to assume it.

File: example/stochastic-depth/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+Stochastic Depth
+================
+
+This folder contains examples showing implementation of the stochastic depth algorithm described in the paper
+Huang, Gao, et al. ["Deep networks with stochastic depth."](https://arxiv.org/abs/1603.09382)
+arXiv preprint arXiv:1603.09382 (2016). This paper introduces a new way to perturb networks during training
+in order to improve their performance. Stochastic Depth (SD) is a method for residual networks,
+which randomly removes/deactivates residual blocks
+during training.
```
Comment (sandeep-krishnamurthy) [inline example/stochastic-depth/README.md]: nit: unintended new line?

File: example/vae/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- [Jupyter Notebook](http://jupyter.org/index.html)
+- Matplotlib
+
+Files in this folder:
+- **VAE_example.ipynb** : Jupyter notebook which explains concept of VAE step by step and also shows how to use
+MXNet-based VAE class(from VAE.py) to do the training directly.
+
+- **VAE.py** : Contains class which implements the Variational Auto Encoder. This is used in the above tutorial.
+
+In VAE, the encoder becomes a variational inference network that maps the data to the a distribution
```
Comment (sandeep-krishnamurthy) [inline example/vae/README.md]: to the a => to a disribution

File: example/model-parallel/matrix_factorization/README.md (hunk 4 above)
Comment (Roshrini) [reply example/model-parallel/matrix_factorization/README.md]: This example shows usage of one part of the model to be trained on CPU and other on GPU. so its necessary to have gpus on the machine. I added one line saying this:
"it is necessary to have GPUs available on the machine
to run this example."

Comment (sandeep-krishnamurthy) [reply example/model-parallel/matrix_factorization/README.md]: num_gpus is 1 to cover more general user base?

**Pull request merged**